sources/PPU/Background.cpp

```diff
@@ -47,107 +52,50 @@ namespace ComSquare::PPU
 		}
 	}
 
-	void Background::drawBgTile(uint16_t data, Vector2<int> pos)
+	void Background::_drawBgTile(uint16_t data, Vector2<int> pos)
 	{
-		uint16_t graphicAddress;
 		union TileMapData tileData;
-		std::vector<uint16_t> palette;
-		int index = 0;
-		uint8_t reference = 0;
-		uint32_t color = 0;
 
 		tileData.raw = data;
-		palette = getPalette(tileData.palette);
+
+		if (tileData.tilePriority != this->_priority)
+			return;
+
+		uint16_t graphicAddress;
+		Vector2i tileOffset = {0, 0};
 		// X horizontal
 		// Y vertical
-		graphicAddress = this->_tilesetAddress + (tileData.posY * NbTilePerRow * this->_bpp * TileBaseByteSize) + (tileData.posX * this->_bpp * TileBaseByteSize);
+
+		this->_tileRenderer.setPaletteIndex(tileData.palette);
+		for (int i = 0; i < this->_characterNbPixels.y; i += Tile::NbPixelsHeight) {
+			for (int j = 0; j < this->_characterNbPixels.x; j += Tile::NbPixelsWidth) {
+				graphicAddress = this->_tilesetAddress +
+				                 ((tileData.posY + tileOffset.y) * NbTilePerRow * this->_bpp * Tile::BaseByteSize) +
+				                 ((tileData.posX + tileOffset.x) * this->_bpp * Tile::BaseByteSize);
+				this->_tileRenderer.render(graphicAddress);
+				merge2DArray(this->_tileBuffer, this->_tileRenderer.buffer, {j, i});
```

Comment:
Put this in a static class or a namespace and call it explicitly. We don't really understand where they are now.

Reply:
fixed